Bump Istio to 1.24 (#1149)
kyma-project/istio: docs/release-notes/1.12.0.md

## New Features

### Istio version update to 1.24.1 

| **Component** | **Version** |
|---------------|-------------|
| Istio         | 1.24.1      |
| Envoy         | 1.32.2      |

We've updated the Istio version to 1.24.1 ([#1024](https://github.com/kyma-project/istio/pull/1024)).
Read the Istio 1.24.1 [release announcement](https://istio.io/latest/news/releases/1.24.x/announcing-1.24.1/)
and [Istio 1.24.0 Change Notes](https://istio.io/latest/news/releases/1.24.x/announcing-1.24/change-notes/).

> This version change involves the following changes to compatibility version parameters:

Reviewer: ```suggestion
This version includes the following changes to compatibility version parameters.
```

> Previous compatibility version parameters switching from false to true:

Reviewer: ```suggestion
The parameters for the previous compatibility version are switched from 'false' to 'true':
```

| Parameter                            | Description                                                                                                                                                                         |
|--------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | ENABLE\_DELIMITED\_STATS\_TAG\_REGEX | If true, pilot will use the new delimited stat tag regex to generate Envoy stats tags.                                                                                              |
> | ENABLE\_DEFERRED\_CLUSTER\_CREATION  | If enabled, Istio will create clusters only when there are requests. This will save memory and CPU cycles in cases where there are lots of inactive clusters and > 1 worker thread. |

Reviewer: ```suggestion
| ENABLE\_DELIMITED\_STATS\_TAG\_REGEX | If true, pilot uses the new delimited stat tag regex to generate Envoy stats tags.                                                                                              |
| ENABLE\_DEFERRED\_CLUSTER\_CREATION  | If enabled, Istio creates clusters only when there are requests. This saves memory and CPU cycles in cases where there are lots of inactive clusters and more than one worker thread. |
```

> New compatibility version parameters switching set to false:

Reviewer: ```suggestion
This compatibility version applies the new parameters set to `false`:
```

| Parameter                                             | Description                                                                                                                                        |
|-------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------|

> | ENABLE\_INBOUND\_RETRY\_POLICY       | If true, enables retry policy for inbound routes which automatically retries requests that were reset before it reaches the service.                                                |

Reviewer: ```suggestion
| ENABLE\_INBOUND\_RETRY\_POLICY       | If true, enables retry policy for inbound routes, which automatically retries any requests that were reset before reaching the intended service.                                                |
```

> | EXCLUDE\_UNSAFE\_503\_FROM\_DEFAULT\_RETRY            | If true, excludes unsafe retry on 503 from default retry policy.                                                                                   |

Reviewer: ```suggestion
| EXCLUDE\_UNSAFE\_503\_FROM\_DEFAULT\_RETRY            | If true, excludes unsafe retry on `503` from the default retry policy.                                                                                   |
```

> | PREFER\_DESTINATIONRULE\_TLS\_FOR\_EXTERNAL\_SERVICES | If true, external services will prefer the TLS settings from DestinationRules over the metadata TLS settings.                                      |

Reviewer: ```suggestion
| PREFER\_DESTINATIONRULE\_TLS\_FOR\_EXTERNAL\_SERVICES | If true, external services prefer the TLS settings from DestinationRules over the metadata TLS settings.                                      |
```

> | ENABLE\_ENHANCED\_DESTINATIONRULE\_MERGE              | If enabled, Istio merge DestinationRules considering their exportTo fields, they will be kept as independent rules if the exportTos are not equal. |

Reviewer: ```suggestion
| ENABLE\_ENHANCED\_DESTINATIONRULE\_MERGE              | If enabled, Istio merges DestinationRules considering their **exportTo** fields. The DestinationRules are then kept as independent rules if the **exportTo** fields are not equal. |
```

> | PILOT\_UNIFIED\_SIDECAR\_SCOPE                        | If true, unified SidecarScope creation will be used. This is only intended as a temporary feature flag for backwards compatibility.                |

Reviewer: ```suggestion
| PILOT\_UNIFIED\_SIDECAR\_SCOPE                        | If true, unified SidecarScope creation is used. This is only intended as a temporary feature flag for backward compatibility.                |
```

> | ENABLE\_DEFERRED\_STATS\_CREATION                     | If enabled, Istio will lazily initialize a subset of the stats.                                                                                    |

Reviewer: ```suggestion
| ENABLE\_DEFERRED\_STATS\_CREATION                     | If enabled, Istio lazily initializes a subset of the stats.                                                                                    |
```

> | BYPASS\_OVERLOAD\_MANAGER\_FOR\_STATIC\_LISTENERS     | If enabled, overload manager will not be applied to static listeners.                                                                              |

Reviewer: ```suggestion
| BYPASS\_OVERLOAD\_MANAGER\_FOR\_STATIC\_LISTENERS     | If enabled, the overload manager is not applied to static listeners.                                                                              |
```
